### DISCUSSION TITLE
Prevent user to add new blocks

### DISCUSSION THREAD

**Discussion by mparfait** [codex-team/editor.js#1899] [Q&A]
Hi everyone,

Is there already a way to prevent user to add a new blocks.
For example, if I decide to fill the editor with only a block header and a block paragraph, I would like to restrict the user to only edit the header and the paragraph and not to add others headers or paragraphs.

Example here : 
`
   const editor = new EditorJS({
    holderId : 'editorjs',
    tools: {
      header: {
        class: Header,
        config: {
          placeholder: 'Enter a header'
        }
      }
    },
    
    data : {
      blocks: [
        {
            type: "header",
            data: {
              text: "Mon titre"
            }
          },
          {
            type: "paragraph",
            data: {
              text: "Mon texte"
            }
          }          
        ],
    }

  });
`
Thanks !

**Comment by GruFFix** [comment]
Did you find how to do it?

**Comment by mparfait** [comment]
Hello GruFFix,
No, I've tested this tools during an R&D period and I left this project a few more days after this ticket. Maybe I will get back into this but I don't know when.

**Comment by CavalcanteLeo** [comment]
i'm facing the same issue, any news?

**Comment by elfeffe** [comment]
Is there any news about this? I also need to let my users add one header, and nothing else.

@khaydarov Can you help?

**Comment by xAzoom** [comment]
Any news?

**Comment by vkarisari** [comment]
need single line/block editor too....
